### PR TITLE
Documentation for Preview Dataset

### PR DESCRIPTION
**Pull request by rashidakanchwala**
## Description

Documentation for the new changes in 'Preview Datasets'

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: docs/source/preview_datasets.md [hunk 1]
```diff
@@ -1,35 +1,48 @@
-# Preview data in Kedro-Viz
+# Preview datasets in Kedro-Viz
 
-This page describes how to preview data from different datasets in a Kedro project with Kedro-Viz. Dataset preview was introduced in Kedro-Viz version 6.3.0, which offers preview for `CSVDatasets` and `ExcelDatasets`.
+To provide users with a glimpse of their datasets within a Kedro project, Kedro-Viz offers a preview feature. This feature was introduced in Kedro-Viz version 6.3.0 and expanded upon in version 8.0.0. Initially, it supported `CSVDatasets` and `ExcelDatasets`, and later extended to encompass additional dataset types such as `PlotlyDatasets` and image datasets like `MatplotlibWriter`.
```
Comment (ravi-kumar-pilla) [inline docs/source/preview_datasets.md:3]: [Nit] `ExcelDatasets`, and **was** later extended to ..

File: docs/source/preview_datasets.md [hunk 2]
```diff
@@ -1,35 +1,48 @@
...
-If you haven't installed Kedro {doc}`follow the documentation to get set up<kedro:get_started/install>`. 
+1. **TablePreview:** For datasets returning tables/dataframes.
+2. **JSONPreview:** For datasets returning JSON objects.
+3. **PlotlyPreview:** For datasets returning Plotly JSON objects.
+4. **ImagePreview:** For datasets returning base64-encoded image strings.
 
-```{important}
-We recommend that you use the same version of Kedro that was most recently used to test this tutorial (0.19.0). To check the version installed, type `kedro -V` in your terminal window.
-```
+Whilst we currently support the above datasets. We are soon going to extend this functionality to other datasets. Users with custom datasets can also extend the preview functionality and we will cover that in the following sections.
```
Comment (ravi-kumar-pilla) [inline docs/source/preview_datasets.md:12]: While we currently support the aforementioned **datasets, we** are soon going to extend this functionality to include other datasets. Users with custom datasets can also expand the preview functionality, and we will cover that in the following sections.

File: docs/source/preview_datasets.md [hunk 3]
```diff
@@ -1,35 +1,48 @@
...
 
-When prompted for a project name, you can enter anything, but we will assume `Spaceflights` throughout.
+**Preview Matplotlib Charts**
 
-When your project is ready, navigate to the root directory of the project.
+The page titled [Preview Matplotlib charts in Kedro-viz](./preview_matplotlib_datasets.md) contains a spaceflight tutorial that explains how you can create static visualizations using `MatplotlibWriterDataset` on Kedro-viz.
+
+**Extend Preview to Custom Datasets**
+
+The page titled [Extend Preview to Custom Datasets](./preview_custom_datasets.md) contains information on how you can set up preview for custom datasets and what types are supported by Kedro-viz.
```
Comment (ravi-kumar-pilla) [inline docs/source/preview_datasets.md:29]: The page titled [Extend Preview to Custom Datasets](./preview_custom_datasets.md) contains information on **how to set up previews for custom datasets and which types are supported by Kedro-Viz**

File: docs/source/preview_datasets.md [hunk 4]
```diff
@@ -1,35 +1,48 @@
...
 
-Kedro-Viz version 6.3.0 currently supports preview of two types of datasets:
 
-* `pandas.CSVDataset`
-* `pandas.ExcelDataset`
+## Disabling Previews
 
 
-To enable dataset preview, add the `preview_args` attribute to the kedro-viz configuration under the `metadata` section in the Data Catalog. Within preview_args, specify `nrows` as the number of rows to preview for the dataset.
+To disable dataset previews for specific datasets, you need to set preview: false under the kedro-viz key within the metadata section of your conf.yml file. Here's an example configuration:
```
Comment (ravi-kumar-pilla) [inline docs/source/preview_datasets.md:45]: should we mention that previewing is made default from latest version of viz ? If someone is using old version of viz, it will be opt in ? Also, it will be a good idea to mention about the kedro-datasets version this new feature of kedro viz supports somewhere in the doc. Thank you

Comment (rashidakanchwala) [reply docs/source/preview_datasets.md:45]: good point.

File: docs/source/preview_datasets.md [hunk 5]
```diff
@@ -1,84 +1,62 @@
...
+:hidden:
+preview_matplotlib_datasets
+preview_plotly_datasets
+preview_pandas_datasets
+preview_custom_datasets
 ```
 
 
 
-## Previewing Data on Kedro-viz
```
Comment (rashidakanchwala) [inline docs/source/preview_datasets.md:65]: this entire section has now moved to `Preview Tabular Data on Kedro-viz`

File: docs/source/preview_pandas_datasets.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
+# Preview Tabular Data in Kedro-viz
```
Comment (rashidakanchwala) [inline docs/source/preview_pandas_datasets.md:1]: This content in this section is not new and it is just moved to a new page. Earlier it was a part of the 'Preview Datasets' page

**Review by merelcht — COMMENTED** [review]
Some initial comments, but I'll have another look next week 🙂

File: docs/source/preview_custom_datasets.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+The return type of the `preview()` function should match one of the following types, as defined in the `kedro-datasets` source code (_typing.py file):
+
+```python
+TablePreview = NewType("TablePreview", dict)
+ImagePreview = NewType("ImagePreview", bytes)
+PlotlyPreview = NewType("PlotlyPreview", dict)
+JSONPreview = NewType("JSONPreview", dict)
+```
+
+Below is an example demonstrating how to implement the preview() function for a CustomDataset class that utilizes TablePreview to enable previewing tabular data on Kedro-viz:
```
Comment (noklam) [inline docs/source/preview_custom_datasets.md:14]:
```suggestion
Below is an example demonstrating how to implement the preview() function for a CustomDataset class that utilizes `TablePreview` to enable previewing tabular data on Kedro-viz:
```

File: docs/source/preview_custom_datasets.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+When creating a custom dataset, if you wish to enable data preview for that dataset, you must implement a `preview()` function within the custom dataset class. Kedro-viz currently supports previewing Tables, Plotly charts, Images, and JSON objects.
+
+The return type of the `preview()` function should match one of the following types, as defined in the `kedro-datasets` source code (_typing.py file):
+
+```python
+TablePreview = NewType("TablePreview", dict)
+ImagePreview = NewType("ImagePreview", bytes)
+PlotlyPreview = NewType("PlotlyPreview", dict)
+JSONPreview = NewType("JSONPreview", dict)
+```
```
Comment (noklam) [inline docs/source/preview_custom_datasets.md:12]: Can we show an example for each type? This help users to link what are they expecting to see in the UI.

https://clear.ml/docs/latest/docs/guides/reporting/plotly_reporting

File: docs/source/preview_datasets.md [hunk 9]
```diff
@@ -1,84 +1,62 @@
-# Preview data in Kedro-Viz
+# Preview datasets in Kedro-Viz
 
-This page describes how to preview data from different datasets in a Kedro project with Kedro-Viz. Dataset preview was introduced in Kedro-Viz version 6.3.0, which offers preview for `CSVDatasets` and `ExcelDatasets`.
+To provide users with a glimpse of their datasets within a Kedro project, Kedro-Viz offers a preview feature. This feature was introduced in Kedro-Viz version 6.3.0 and expanded upon in version 8.0.0. Initially, it supported `CSVDatasets` and `ExcelDatasets`, and was later extended to encompass additional dataset types such as `PlotlyDatasets` and image datasets like `MatplotlibWriter`.
```
Comment (noklam) [inline docs/source/preview_datasets.md:3]:
```suggestion
To provide users with a glimpse of their datasets within a Kedro project, Kedro-Viz offers a preview feature. This feature was introduced in Kedro-Viz version 6.3.0 and expanded upon in version 8.0.0. Initially, it supported `CSVDataset` and `ExcelDataset`, and was later extended to encompass additional dataset types such as `PlotlyDataset` and image datasets like `MatplotlibWriter`.
```

**Review by noklam — COMMENTED** [review]
A few minor comments, I think the most important one is an example of CustomDataset. Other than that there are quite a few inconsistency uses of `Dataset` `Datasets` and `DataSet`.

File: docs/source/preview_pandas_datasets.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+  filepath: data/01_raw/shuttles.xlsx
+  metadata:
+    kedro-viz:
+      layer: raw
+      preview_args:
+          nrows: 15
+```
+
+If no preview_args are specified, the default preview will show the first 5 rows.
+
```
Comment (noklam) [inline docs/source/preview_pandas_datasets.md:53]: This was asked in the Slack once, I think we should make it obvious that the `preview_args` is the argument that get pass into the `preview` function directly, and user can have arbitary arguments.

```python
def preview(self, arg1, arg2):
  ...
```

File: docs/source/preview_custom_datasets.md [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+
+Below is an example demonstrating how to implement the preview() function for a CustomDataset class that utilizes TablePreview to enable previewing tabular data on Kedro-viz:
+
+```python 
+
+from kedro_datasets._typing import TablePreview
+
+class CustomDataset:
+    def preview(self, nrows: int = 5) -> TablePreview:
```
[the remark this answers is not in the document]
Comment (rashidakanchwala) [reply docs/source/preview_custom_datasets.md:21]: I will need some help on this as I can't think of a realistic CustomDataset that is not a part of kedr-datasets. @astrojuanlu  -- do you have some ideas?

File: docs/source/preview_pandas_datasets.md (hunk 10 above)
Comment (rashidakanchwala) [reply docs/source/preview_pandas_datasets.md:53]: So I think for pandas datasets .. it is specific to `nrows` as we wrote the `preview()` func. 

But I have updated the custom dataset docs to include arguments. Thanks for highlighting this @noklam

File: docs/source/preview_custom_datasets.md [hunk 12]
```diff
@@ -0,0 +1,52 @@
+# Extend Preview to Custom Datasets
```
Comment (astrojuanlu) [inline docs/source/preview_custom_datasets.md:1]:
```suggestion
# Extend preview to custom datasets
```

as per https://github.com/kedro-org/kedro/wiki/Kedro-documentation-style-guide#capitalisation

File: docs/source/preview_custom_datasets.md [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+
+When creating a custom dataset, if you wish to enable data preview for that dataset, you must implement a `preview()` function within the custom dataset class. Kedro-Viz currently supports previewing tables, Plotly charts, images, and JSON objects.
+
+The return type of the `preview()` function should match one of the following types, as defined in the `kedro-datasets` source code (_typing.py file):
```
Comment (astrojuanlu) [inline docs/source/preview_custom_datasets.md:5]: Any chance to add a link to such file?

File: docs/source/preview_datasets.md [hunk 14]
```diff
@@ -1,84 +1,62 @@
...
-```{important}
-We recommend that you use the same version of Kedro that was most recently used to test this tutorial (0.19.0). To check the version installed, type `kedro -V` in your terminal window.
-```
+1. **TablePreview:** For datasets returning tables/dataframes.
+2. **JSONPreview:** For datasets returning JSON objects.
+3. **PlotlyPreview:** For datasets returning Plotly JSON objects.
+4. **ImagePreview:** For datasets returning base64-encoded image strings.
 
-In your terminal window, navigate to the folder you want to store the project. Generate the spaceflights tutorial project with all the code in place by using the [Kedro starter for the spaceflights tutorial](https://github.com/kedro-org/kedro-starters/tree/main/spaceflights-pandas):
+While we currently support the aforementioned datasets, we are soon going to extend this functionality to include other datasets. Users with custom datasets can also expand the preview functionality, , which is covered in the section [Extend Preview to Custom Datasets](./preview_custom_datasets.md).
```
Comment (astrojuanlu) [inline docs/source/preview_datasets.md:13]: Do we have Vale enabled on this repo? I get the sense that _aforementioned_ would be flagged as too wordy

File: docs/source/preview_datasets.md [hunk 15]
```diff
@@ -1,84 +1,62 @@
...
 
+```{note}
+Starting from Kedro-Viz version 8.0.0, the preview functionality on Kedro-Viz is now opt-out. For previous versions, this is an opt-in mechanism and users are required to specify `preview-args` for the preview to be enabled.
 
-```bash
-kedro new --starter=spaceflights-pandas
+By default, preview is now enabled for datasets. If you wish to disable preview for datasets, please refer to the [Disable Preview section](./preview_datasets.md#disabling-previews) for instructions.
```
Comment (astrojuanlu) [inline docs/source/preview_datasets.md:18]: What about

"Starting from Kedro-Viz 8.0.0 preview is now enabled for all datasets. If you wish to disable it for a specific dataset, refer to the [Disable Preview section](./preview_datasets.md#disabling-previews) for instructions."

File: docs/source/preview_datasets.md [hunk 16]
```diff
@@ -1,84 +1,62 @@
...
-```yaml
-companies:
-  type: pandas.CSVDataSet
-  filepath: data/01_raw/companies.csv
-  metadata:
-    kedro-viz:
-      layer: raw
-      preview_args:
-          nrows: 5
+The page titled [Preview Tabular Data in Kedro-Viz](./preview_pandas_datasets.md) contains a spaceflight tutorial that explains how you can enable preview on Tabular datasets such as `pandas.CSVDataset` and `pandas.ExcelDataset`.
```
Comment (astrojuanlu) [inline docs/source/preview_datasets.md:23]:
```suggestion
See [Preview tabular data in Kedro-Viz](./preview_pandas_datasets.md) for an guide on how you can enable preview on tabular datasets such as `pandas.CSVDataset` and `pandas.ExcelDataset`.
```

File: docs/source/preview_datasets.md [hunk 17]
```diff
@@ -1,84 +1,62 @@
...
-shuttles:
-  type: pandas.ExcelDataSet
-  filepath: data/01_raw/shuttles.xlsx
-  metadata:
-    kedro-viz:
-      layer: raw
-      preview_args:
-          nrows: 15
-```
+The page titled [Preview Plotly charts in Kedro-Viz](./preview_plotly_datasets.md) contains a spaceflight tutorial that explains how you can create interactive visualizations using `PlotlyDataset` on Kedro-Viz.
```
Comment (astrojuanlu) [inline docs/source/preview_datasets.md:27]:
```suggestion
See [Preview Plotly charts in Kedro-Viz](./preview_plotly_datasets.md) for a guide on how you can create interactive visualizations using `PlotlyDataset` on Kedro-Viz.
```

File: docs/source/preview_datasets.md [hunk 18]
```diff
@@ -1,84 +1,62 @@
...
 
+**Preview Matplotlib Charts**
 
+The page titled [Preview Matplotlib charts in Kedro-Viz](./preview_matplotlib_datasets.md) contains a spaceflight tutorial that explains how you can create static visualizations using `MatplotlibWriterDataset` on Kedro-Viz.
```
Comment (astrojuanlu) [inline docs/source/preview_datasets.md:31]: (As above)

File: docs/source/preview_datasets.md [hunk 19]
```diff
@@ -1,84 +1,62 @@
...
 
-## Previewing Data on Kedro-viz
+**Extend Preview to Custom Datasets**
 
-After you've configured the Data Catalog, you can preview the datasets on Kedro-Viz. Start Kedro-Viz by running the following command in your terminal:
+The page titled [Extend Preview to Custom Datasets](./preview_custom_datasets.md) contains information on how to set up previews for custom datasets and which types are supported by Kedro-Viz.
```
Comment (astrojuanlu) [inline docs/source/preview_datasets.md:35]: (As above)

File: docs/source/preview_datasets.md [hunk 20]
```diff
@@ -1,84 +1,62 @@
...
-
-Click on each dataset node to see a small preview in the metadata panel:
 
 
-![](./images/preview_datasets_metadata.png)
+## Disabling Previews
 
 
-View the larger preview of the dataset by clicking the `Expand Preview Table` button on the bottom of the metadata panel.
+To disable dataset previews for specific datasets, you need to set `preview: false` under the `kedro-viz` key within the `metadata` section of your `conf.yml` file. Here's an example configuration:
```
Comment (astrojuanlu) [inline docs/source/preview_datasets.md:51]:
```suggestion
To disable dataset previews for specific datasets, you need to set `preview: false` under the `kedro-viz` key within the `metadata` section of your `catalog.yml` file. Here's an example configuration:
```

File: docs/source/preview_matplotlib_datasets.md [hunk 21]
```diff
@@ -1,10 +1,10 @@
-# Visualise charts in Kedro-Viz with Matplotlib
+# Preview Matplotlib charts in Kedro-Viz 
 
-This page describes how to output interactive visualisations of a Kedro project with Kedro-Viz, which supports integration with [Matplotlib](https://matplotlib.org/). You can view Matplotlib charts in Kedro-Viz when you use the MatplotLibWriter dataset.
+This page describes how to output static visualisations of a Kedro project with Kedro-Viz, which supports integration with [Matplotlib](https://matplotlib.org/). You can view Matplotlib charts in Kedro-Viz when you use the MatplotLibWriter dataset.
```
Comment (astrojuanlu) [inline docs/source/preview_matplotlib_datasets.md:3]:
```suggestion
This page describes how to output static visualisations of a Kedro project with Kedro-Viz, which supports integration with [Matplotlib](https://matplotlib.org/). You can view Matplotlib charts in Kedro-Viz when you use the MatplotlibWriter dataset.
```

plus a link to https://docs.kedro.org/projects/kedro-datasets/en/kedro-datasets-2.0.0/api/kedro_datasets.matplotlib.MatplotlibWriter.html

File: docs/source/preview_matplotlib_datasets.md [hunk 22]
```diff
@@ -1,10 +1,10 @@
...
 
 
 ```{note}
-The MatplotlibWriter dataset converts Matplotlib objects to image files. This means that Matplotlib charts within Kedro-Viz are static and not interactive, unlike the [Plotly charts seen separately](./visualise_charts_with_plotly.md).
+The MatplotlibWriter dataset converts Matplotlib objects to image files. This means that Matplotlib charts within Kedro-Viz are static and not interactive, unlike the [Plotly charts seen separately](./preview_plotly_datasets.md).
```
Comment (astrojuanlu) [inline docs/source/preview_matplotlib_datasets.md:7]:
```suggestion
The `MatplotlibWriter` dataset converts Matplotlib objects to image files. This means that Matplotlib charts within Kedro-Viz are static and not interactive, unlike the [Plotly charts seen separately](./preview_plotly_datasets.md).
```

File: docs/source/preview_pandas_datasets.md [hunk 23]
```diff
@@ -0,0 +1,74 @@
...
+We use the {doc}`spaceflights tutorial<kedro:tutorial/spaceflights_tutorial>` to demonstrate how to add data preview for the `customer`, `shuttle` and `reviews` datasets. Even if you have not yet worked through the tutorial, you can still follow this example; you'll need to use the Kedro starter for the spaceflights tutorial to generate a copy of the project with working code in place.
+
+If you haven't installed Kedro {doc}`follow the documentation to get set up<kedro:get_started/install>`. 
+
+```{important}
+We recommend that you use the same version of Kedro that was most recently used to test this tutorial (0.19.0). To check the version installed, type `kedro -V` in your terminal window.
+```
+
+In your terminal window, navigate to the folder you want to store the project. Generate the spaceflights tutorial project with all the code in place by using the [Kedro starter for the spaceflights tutorial](https://github.com/kedro-org/kedro-starters/tree/main/spaceflights-pandas):
```
Comment (astrojuanlu) [inline docs/source/preview_pandas_datasets.md:11]: Probably capitalise Spaceflights?

```suggestion
In your terminal window, navigate to the folder you want to store the project. Generate the Spaceflights tutorial project with all the code in place by using the [Kedro starter for the spaceflights tutorial](https://github.com/kedro-org/kedro-starters/tree/main/spaceflights-pandas):
```

File: docs/source/preview_pandas_datasets.md [hunk 24]
```diff
@@ -0,0 +1,74 @@
...
+
+
+```bash
+kedro new --starter=spaceflights-pandas
+```
+
+When prompted for a project name, you can enter anything, but we will assume `Spaceflights` throughout.
```
Comment (astrojuanlu) [inline docs/source/preview_pandas_datasets.md:18]:
```suggestion
When prompted for a project name, you can enter anything, but we will assume Spaceflights throughout.
```

File: docs/source/preview_pandas_datasets.md [hunk 25]
```diff
@@ -0,0 +1,74 @@
...
+
+When your project is ready, navigate to the root directory of the project.
+
+## Configure the Data Catalog
+
+Kedro-Viz version 8.0.0 supports previews for two types of tabular datasets: pandas.CSVDataset and pandas.ExcelDataset. Previews are enabled by default, showing the first 5 rows unless otherwise specified using preview_args.
```
Comment (astrojuanlu) [inline docs/source/preview_pandas_datasets.md:24]:
```suggestion
Kedro-Viz version 8.0.0 supports previews for two types of tabular datasets: `pandas.CSVDataset` and `pandas.ExcelDataset`. Previews are enabled by default, showing the first 5 rows unless otherwise specified using `preview_args`.
```

ideally with links to the docs https://docs.kedro.org/projects/kedro-datasets/en/kedro-datasets-2.0.0/api/kedro_datasets.pandas.CSVDataset.html

File: docs/source/preview_pandas_datasets.md [hunk 26]
```diff
@@ -0,0 +1,74 @@
...
+
+Example configuration in catalog.yml:
```
Comment (astrojuanlu) [inline docs/source/preview_pandas_datasets.md:26]:
```suggestion
Example configuration in `catalog.yml`:
```

File: docs/source/preview_pandas_datasets.md [hunk 27]
```diff
@@ -0,0 +1,74 @@
...
+  type: pandas.ExcelDataset
+  filepath: data/01_raw/shuttles.xlsx
+  metadata:
+    kedro-viz:
+      layer: raw
+      preview_args:
+          nrows: 15
+```
+
+If no preview_args are specified, the default preview will show the first 5 rows.
```
Comment (astrojuanlu) [inline docs/source/preview_pandas_datasets.md:52]:
```suggestion
If no `preview_args` are specified, the default preview will show the first 5 rows.
```

File: docs/source/preview_custom_datasets.md (hunk 13 above)
Comment (rashidakanchwala) [reply docs/source/preview_custom_datasets.md:5]: is the github link fine ? - https://github.com/kedro-org/kedro-plugins/blob/main/kedro-datasets/kedro_datasets/_typing.py 

I can't seem to find docs source code link for _typing.py

File: docs/source/preview_datasets.md [hunk 28]
```diff
@@ -1,84 +1,62 @@
-# Preview data in Kedro-Viz
+# Preview datasets in Kedro-Viz
 
-This page describes how to preview data from different datasets in a Kedro project with Kedro-Viz. Dataset preview was introduced in Kedro-Viz version 6.3.0, which offers preview for `CSVDatasets` and `ExcelDatasets`.
+To provide users with a glimpse of their datasets within a Kedro project, Kedro-Viz offers a preview feature. This feature was introduced in Kedro-Viz version 6.3.0 and expanded upon in version 8.0.0. Initially, it supported `CSVDataset` and `ExcelDataset`, and was later extended to encompass additional dataset types such as `PlotlyDataset` and image datasets like `MatplotlibWriter`.
```
[the remark this answers is not in the document]
Comment (rashidakanchwala) [reply docs/source/preview_datasets.md:3]: that makes sense.

File: docs/source/preview_datasets.md (hunk 14 above)
Comment (rashidakanchwala) [reply docs/source/preview_datasets.md:13]: i don't think so.

File: docs/source/preview_pandas_datasets.md (hunk 23 above)
Comment (rashidakanchwala) [reply docs/source/preview_pandas_datasets.md:11]: I noticed it's mostly lower-case elsewhere in the docs so I will leave it as is.

File: docs/source/preview_pandas_datasets.md (hunk 24 above)
Comment (rashidakanchwala) [reply docs/source/preview_pandas_datasets.md:18]: not sure about this one, because we are mentioning as project_name.

File: docs/source/preview_custom_datasets.md (hunk 13 above)
Comment (astrojuanlu) [reply docs/source/preview_custom_datasets.md]: Yeah this is not documented, so a link to the source code is fine for now.

**Review by astrojuanlu — APPROVED** [review]
LGTM!

File: docs/source/preview_custom_datasets.md [hunk 29]
```diff
@@ -0,0 +1,59 @@
...
+
+from kedro_datasets._typing import TablePreview
+
+class CustomDataset:
+    def preview(self, nrows, ncolumns, filters) -> TablePreview:
+        # Add logic for generating preview
+```
+
+
+## Examples of Previews
```
Comment (merelcht) [inline docs/source/preview_custom_datasets.md:41]: Is there way to show the `JSONPreview` as well?

File: docs/source/preview_custom_datasets.md [hunk 30]
```diff
@@ -0,0 +1,59 @@
...
+        filters: name 
+```
+
+```python 
+
+from kedro_datasets._typing import TablePreview
+
+class CustomDataset:
+    def preview(self, nrows, ncolumns, filters) -> TablePreview:
+        # Add logic for generating preview
```
Comment (merelcht) [inline docs/source/preview_custom_datasets.md:37]: When I suggested adding a working example, I didn't necessarily mean anything complex, just in this case some code that would produce a working `TablePreview` and demonstrates how `nrows`, `ncolumns` and `filters` would be used.

File: docs/source/preview_pandas_datasets.md [hunk 31]
```diff
@@ -0,0 +1,74 @@
...
+
+We use the {doc}`spaceflights tutorial<kedro:tutorial/spaceflights_tutorial>` to demonstrate how to add data preview for the `customer`, `shuttle` and `reviews` datasets. Even if you have not yet worked through the tutorial, you can still follow this example; you'll need to use the Kedro starter for the spaceflights tutorial to generate a copy of the project with working code in place.
+
+If you haven't installed Kedro {doc}`follow the documentation to get set up<kedro:get_started/install>`. 
+
+```{important}
+We recommend that you use the same version of Kedro that was most recently used to test this tutorial (0.19.0). To check the version installed, type `kedro -V` in your terminal window.
+```
+
+In your terminal window, navigate to the folder you want to store the project. Generate the spaceflights tutorial project with all the code in place by using the [Kedro starter for the spaceflights tutorial](https://github.com/kedro-org/kedro-starters/tree/main/spaceflights-pandas):
```
Comment (merelcht) [inline docs/source/preview_pandas_datasets.md:11]:
```suggestion
In your terminal window, navigate to the folder where you want to store the project. Generate the spaceflights tutorial project with all the code in place by using the [Kedro starter for the spaceflights tutorial](https://github.com/kedro-org/kedro-starters/tree/main/spaceflights-pandas):
```

File: docs/source/preview_pandas_datasets.md [hunk 32]
```diff
@@ -0,0 +1,74 @@
...
+    kedro-viz:
+      layer: raw
+      preview_args:
+          nrows: 15
+```
+
+If no `preview_args` are specified, the default preview will show the first 5 rows.
+
+
+## Previewing Data on Kedro-Viz
```
Comment (merelcht) [inline docs/source/preview_pandas_datasets.md:55]:
```suggestion
## Previewing data on Kedro-Viz
```

**Review by merelcht — APPROVED** [review]
I left some minor comments and a suggestion on adding an example `preview()` method. It's all non blocking though, so I'll approve and let you decide on whether to add it or not 🙂

File: docs/source/preview_custom_datasets.md (hunk 29 above)
Comment (rashidakanchwala) [reply docs/source/preview_custom_datasets.md:41]: so, the JSON preview is actually experiment tracking oriented hence I was hesitant to share. In the next couple of sprints, we will enable preview for a JSONDataset and then I could add that example then.

File: demo-project/conf/base/catalog_01_raw.yml [hunk 33]
```diff
@@ -3,18 +3,16 @@ companies:
...
 
 reviews:
   type: pandas.CSVDataset
   filepath: ${_base_location}/01_raw/reviews.csv
   metadata:
     kedro-viz:
       layer: raw
-      preview_args:
-        nrows: 10
+      prvoew
```
Comment (ravi-kumar-pilla) [inline demo-project/conf/base/catalog_01_raw.yml:15]: was this intentional change for testing ?

Comment (rashidakanchwala) [reply demo-project/conf/base/catalog_01_raw.yml:15]: this was a mistake :)  .. thanks for flagging
